### DISCUSSION TITLE
Using `escaped_transform` is error-prone

### DISCUSSION THREAD

**Discussion by Zoybean** [winnow-rs/winnow#503] [Q&A]
I've found the documentation for `escaped_transform` to be a bit lacking - it seems to have several constraints on its correct usage, that aren't documented, and weren't obvious to me (though I have little experience with writing parsers):
- It was only through accidentally finding the [`nom` docs for the same function](https://docs.rs/nom/7.1.3/nom/bytes/complete/fn.escaped_transform.html), that I learned that the `Normal` parser must not match the escape character.
- It took trial and error for me to figure out that the `Normal` parser must also not match the empty string. (though the example in the docs helped a bit there - I had to ask, why use `alpha1` and not `alpha0`)

I wasn't sure if a doc improvement counted as a feature request so I figured I'd just open a discussion. This could at least serve as a landing point for folks searching how to use the function.

(similar to #242 but not obviously answered by it)

**Comment by mibmo** [comment]
You just saved me. I also was not aware that it wasn't allowed match `""`, thanks! This should really be in the documentation (same applies to `take_escaped` btw)

**Comment by epage** [comment]
Sorry for the delay in getting to this.  I got overwhelmed with my inbox and I'm slowly whittling it back down.

#568 will improve the docs.  I am tempted to make the empty string case an assert, like `repeat`, see #569.

> I wasn't sure if a doc improvement counted as a feature request so I figured I'd just open a discussion. 

Usability problems, which include documentation, are features / bugs and Issues can be opened or them.

**Reply by mibmo** [reply]
> Sorry for the delay in getting to this. I got overwhelmed with my inbox and I'm slowly whittling it back down.
> 
> #568 will improve the docs. I am tempted to make the empty string case an assert, like `repeat`, see #569.
> 
> > I wasn't sure if a doc improvement counted as a feature request so I figured I'd just open a discussion.
> 
> Usability problems, which include documentation, are features / bugs and Issues can be opened or them.

Hey, no worries! Thanks for the amazing library. It's always okay if you need to take a break :) If you want, I can handle the assert PR (if you want the PR, that is! :)

**Reply by epage** [reply]
> If you want, I can handle the assert PR (if you want the PR, that is! :)

Not ready for making a breaking change at this time.  I'm also unsure if its the right decision.  Input on the issue is always welcome!
